### DISCUSSION TITLE
Accessing EntityManager from inside an entity

### DISCUSSION THREAD

**Discussion by edorgeville** [mikro-orm/mikro-orm#2037] [Q&A] [answered]
Hi,
From my understanding, you can access the EntityManager from an entity by doing `wrap(this, true).__em`. This allows, for example, to do `await wrap(this, true).__em.getRepository(MyEntity).count()`.
However, it seems that this only works on already managed entities (fetched from the DB, for example).
If I create a new unmanaged entity, `__em` will be `undefined`, getting a `TypeError: Cannot read property 'getRepository' of undefined`
I made a mini-repo here to illustrate: https://github.com/edorgeville/mikro-orm-access-em-from-entity

What is the correct way to access the EntityManager from inside an entity, or access other repositories?
Thanks in advance!

**Comment by B4nan** [comment, marked as answer]
The `__em` property is internal and is indeed available only on managed entities. In general you should not use it. Only correct way to make this active record like API is via `RequestContext` helper (`RequestContext.getEntityManager()` will give you the contextual EM fork when you use it inside the request handler).

https://mikro-orm.io/docs/identity-map/#requestcontext-helper-for-di-containers
